streams/src/main/java/org/apache/kafka/streams/query/QueryResult.java

```diff
@@ -29,10 +29,10 @@
  */
 public final class QueryResult<R> {
 
-    private final List<String> executionInfo = new LinkedList<>();
     private final FailureReason failureReason;
     private final String failure;
     private final R result;
+    private List<String> executionInfo = new LinkedList<>();
```

Comment:
Why not keep it `final` and use `clear()` and `addAll` instead?

For my own education: what is `executionInfo` (and why is it a `String` list)?

Reply:
Thanks; that would be another way to do it. I'm not sure if that would be clearly better or not, though.

It's for getting more details about how the query was actually executed inside of Streams. Right now, if you request it as part of the query, each store layer will report what it did and how long it took. For runtime queries, you wouldn't want to use it, but I wanted to enable debugging if the cases where query execution seems like it's taking longer than expected. Also, it could be used for tracing, in which every Nth query is run with the execution info on.

It's a list of Strings so that each store layer / operation can just add one "line" of info (like a stack trace), but we don't waste time and memory actually concatenating them with newlines. We considered adding more structure (such as having a field for execution time), but kept it as a string so as not to restrict the kind of "execution information" we might find useful to add in the future.